ocp_resources/resource.py

```diff
@@ -134,9 +134,47 @@ def sub_resource_level(current_class: Any, owner_class: Any, parent_class: Any)
     return None
 
 
-# Exceptions classes
+def replace_key_with_hashed_value(resource_dict: Dict[Any, Any], key_name: str) -> Dict[Any, Any]:
+    """
+    Recursively search a nested dictionary for a given key and changes its value to "******" if found.
+    The function supports two key formats:
+    1. Regular path: "path>to>key"
```

Comment:
You said we have 2 formats, but you didn't explain their differences.

Reply:
Added examples. Please let me know if this works.

Reply:
i do not understand the examples; can you provide real ones? i.e example dict and example path

Reply:
I didn't push the patch yet. I will do that.